tests/db/test_bigchain_api.py

```diff
@@ -7,196 +7,68 @@
-        vote['signature'] = PrivateKey(alice.private_key).sign(vote_data.encode())
-        b.write_vote(vote)
-
-        with pytest.raises(CyclicBlockchainError):
-            b.get_last_voted_block()
-
-    @pytest.mark.genesis
-    def test_try_voting_while_constructing_cyclic_blockchain(self, b,
-                                                             monkeypatch, alice):
-        from bigchaindb.common.exceptions import CyclicBlockchainError
-        from bigchaindb.models import Transaction
-
-        tx = Transaction.create([alice.public_key], [([alice.public_key], 1)])
-        tx = tx.sign([alice.private_key])
-        block1 = b.create_block([tx])
-
-        # We can simply submit twice the same block id and check if `Bigchain`
-        # throws
-        with pytest.raises(CyclicBlockchainError):
-            b.vote(block1.id, block1.id, True)
-
-    @pytest.mark.genesis
-    def test_has_previous_vote_when_already_voted(self, b, monkeypatch, alice):
-        from bigchaindb.models import Transaction
-
-        tx = Transaction.create([alice.public_key], [([alice.public_key], 1)])
-        tx = tx.sign([alice.private_key])
-
-        monkeypatch.setattr('time.time', lambda: 1)
-        block = b.create_block([tx])
-        b.write_block(block)
-
-        assert b.has_previous_vote(block.id) is False
-
-        vote = b.vote(block.id, b.get_last_voted_block().id, True)
-        b.write_vote(vote)
-
-        assert b.has_previous_vote(block.id) is True
-
-    @pytest.mark.genesis
-    def test_get_spent_with_double_inclusion_detected(self, b, monkeypatch, alice):
-        from bigchaindb.exceptions import CriticalDoubleInclusion
+    @pytest.mark.tendermint
+    def test_get_spent_with_double_inclusion_detected(self, b, alice):
         from bigchaindb.models import Transaction
+        from bigchaindb.backend.exceptions import OperationError
 
         tx = Transaction.create([alice.public_key], [([alice.public_key], 1)])
         tx = tx.sign([alice.private_key])
 
-        monkeypatch.setattr('time.time', lambda: 1000000000)
-        block1 = b.create_block([tx])
-        b.write_block(block1)
+        b.store_bulk_transactions([tx])
 
-        monkeypatch.setattr('time.time', lambda: 1000000020)
         transfer_tx = Transaction.transfer(tx.to_inputs(), [([alice.public_key], 1)],
                                            asset_id=tx.id)
         transfer_tx = transfer_tx.sign([alice.private_key])
-        block2 = b.create_block([transfer_tx])
-        b.write_block(block2)
+        b.store_bulk_transactions([transfer_tx])
 
-        monkeypatch.setattr('time.time', lambda: 1000000030)
         transfer_tx2 = Transaction.transfer(tx.to_inputs(), [([alice.public_key], 1)],
                                             asset_id=tx.id)
         transfer_tx2 = transfer_tx2.sign([alice.private_key])
-        block3 = b.create_block([transfer_tx2])
-        b.write_block(block3)
+        with pytest.raises(OperationError):
```

Comment:
Judging by the name and the previous code of the test, we need to test that `BigchainDB.get_spent` raises`CriticalDoubleSpend` here.

Reply:
No, this isn't a test for a double spend. It's testing that you can't have two txs with the same tx_id

Reply:
It's called `test_get_spent_with_double_inclusion_detected` and it was calling `b.get_spent(tx.id, 0)` expecting it to raise `CriticalDoubleInclusion`, which is like double spend that somehow sneaked into the database. Currently, `BigchainDB.get_spent` raises `core_exceptions.CriticalDoubleSpend` and `DoubleSpend`, which is what we by analogy need to test now.